sparse/impl/KokkosSparse_spiluk_numeric_impl.hpp

```diff
@@ -295,49 +336,111 @@ struct IlukWrap {
       KokkosBatched::TeamSetIdentity<member_type>::invoke(team, lget(block));
     }
 
-    // divide. lhs /= rhs
+    // assign
+    template <typename ViewT>
+    KOKKOS_INLINE_FUNCTION void assign(const ViewT &lhs,
+                                       const CBlock &rhs) const {
+      for (size_type i = 0; i < block_size; ++i) {
+        for (size_type j = 0; j < block_size; ++j) {
+          lhs(i, j) = rhs(i, j);
+        }
+      }
+    }
+
+    // divide. lhs /= rhs (lhs = lhs * rhs^-1)
     KOKKOS_INLINE_FUNCTION
-    void divide(const member_type &team, const LBlock &lhs,
-                const UBlock &rhs) const {
+    void divide(const member_type &team, const Block &lhs, const CBlock &rhs,
```

Comment:
Since we already have `KokkosBatched::Side::{Left,Right}`, why not template divide on it?
You can then either specialize for both `Left` and `Right` or you can `if(std::is_same_v<side, {Left,Right}>)` to implement both cases.